source/agora/node/Ledger.d

```diff
@@ -25,17 +26,21 @@ import agora.node.API;
 import vibe.core.log;
 
 import std.algorithm;
+import std.file;
+import std.path;
 
 /// Ditto
 public class Ledger
 {
-    /// data storage for all the blocks,
+    /// Blockstorage for all the blocks,
     /// currently a single contiguous region to
-    /// improve locality of reference
-    private const(Block)[] ledger;
+    private BlockStorage ledger;
```

Comment:
This should ideally be an interface, something like `IBlockStorage`, where in unittest code it uses in-memory storage for blocks, while in production it uses `BlockStorage` that is disk-based.

cc @Geod24 what do you think?

Reply:
Yes